src/components/directives/in-view/test/helpers.ts

```diff
@@ -0,0 +1,117 @@
+/*!
+ * V4Fire Client Core
+ * https://github.com/V4Fire/Client
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/Client/blob/master/LICENSE
+ */
+
+import type { Locator, Page } from 'playwright';
+
+import type { Watcher, WatchHandler, WatchOptions } from 'core/dom/intersection-watcher';
+
+import { Component, Scroll } from 'tests/helpers';
+
+import { TEST_DIV_MARGIN_TOP_PX } from 'components/directives/in-view/test/const';
+
+/**
+ * A handler to pass to v-in-view
+ * @param watcher - The parameter of the watch handler
+ */
+function handler(watcher: Watcher): void {
+	const div = watcher.target;
+
+	const previousValue = parseInt(
+		div.getAttribute('data-test-in-view') ?? '0',
+		10
+	);
+
+	const nextValue = previousValue + 1;
+	watcher.target.setAttribute('data-test-in-view', nextValue.toString());
+}
+
+/**
+ * Returns the value of the watcher call counter stored in given locator.
+ * @param locator - The source locator
+ */
+export async function getWatcherCallsCount(locator: Locator): Promise<number | null> {
+	const storedValue = await locator.getAttribute('data-test-in-view');
+
+	if (storedValue == null) {
+		return null;
+
+	}
+
+	return parseInt(storedValue, 10);
+}
+
+/**
+ * Force adds test handler (which counts calls) to given watch,
+ * or replaces given one if a function/undefined is provided.
+ *
+ * @param watch - A watch parameter to update
+ */
+function addTestHandlerToWatch(
+	watch: CanUndef<WatchHandler | Partial<WatchOptions>>
+): WatchHandler | WatchOptions & { handler: WatchHandler } {
```

Comment:
Пробелы по краям не нужны

```suggestion
): WatchHandler | WatchOptions & {handler: WatchHandler} {
```